<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <title>IDARE-Quickstart Guide by sysbiolux</title>

    <link rel="stylesheet" href="stylesheets/styles.css">
    <link rel="stylesheet" href="stylesheets/github-light.css">
    <meta name="viewport" content="width=device-width">
    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
  </head>
  <body>
    <div class="wrapper">
      <header>
        <h1>IDARE-Quickstart Guide</h1>
        <p></p>

        <p class="view"><a href="https://github.com/sysbiolux/IDARE-QuickStart">View the Project on GitHub <small>sysbiolux/IDARE-QuickStart</small></a></p>


        <ul>
          <li><a href="https://github.com/sysbiolux/IDARE-QuickStart/zipball/master">Download <strong>ZIP File</strong></a></li>
          <li><a href="https://github.com/sysbiolux/IDARE-QuickStart/tarball/master">Download <strong>TAR Ball</strong></a></li>
          <li><a href="https://github.com/sysbiolux/IDARE-QuickStart">View On <strong>GitHub</strong></a></li>
        </ul>
      </header>
      <section>
        <h1>
<a id="idare-quickstart-guide" class="anchor" href="#idare-quickstart-guide" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>IDARE Quickstart guide</h1>

<p>This Repository contains basic sample datasets to use with the IDARE cytoscape application.
The data can be used with the <em>E.coli</em> core network sbml file obtainable from the <a href="https://github.com/sysbiolux/IDARE-QuickStart/tree/master/Data">Data folder</a>, 
or by right-clicking on this <a href="https://github.com/sysbiolux/IDARE-QuickStart/blob/master/Data/ecoli_core_model.xml">Link</a> and selecting "save as".
Also obtain the excel files in this repository for some sample data.</p>

<p>The following files can be downloaded via the repository zip file, or from the <a href="https://github.com/sysbiolux/IDARE-QuickStart/tree/master/Data">Data Folder</a>:<br>
Fold Change of Gene Expression ethanol vs no ethanol.xls - Data from Horinouchi et al.[1] used to calculate Gene expression fold changes between untreated and ethanol treated samples<br>
Metabolomics fold change treatment vs Control.xls - Data from Wang et al[2] for Metabolomics fold changes of treatment vs control samples<br>
Total relative C13 label.xls - Data from Goodarzi et al.[3] for C13 labeling upon feeding with ethanol for a wild type and a ethanol resistant starin. The data is computed to present the total amount of labeled carbons in each metabolite.<br>
Proteomics Time course of ethanol treated wild type.xls - Data from Soufi et al.[4] epresenting the fold change to time 0 of different protein products associated with the genes in the network.  </p>

<p>The Tutorial Contains three parts.<br>
In Part 1, we set up the network, which is necessary for image matching.<br>
In Part 2, we add images based on the data to the network.<br>
In Part 3, we will split the network into parts based on compartments and pathways.  </p>

<h2>
<a id="part-1-setting-up-the-network-for-idare-in-cytoscape" class="anchor" href="#part-1-setting-up-the-network-for-idare-in-cytoscape" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Part 1: Setting up the Network for IDARE in Cytoscape</h2>

<p>In this example, we will set up the <em>E.Coli</em> core network for use with IDARE and add some additional annotations to the network.</p>

<h4>
<a id="11-loading-the-ecoli-core-model" class="anchor" href="#11-loading-the-ecoli-core-model" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>1.1 Loading the E.Coli Core model</h4>

<p>After installing the IDARE app (either from the Cytoscape app store or from <a href="http://idare.uni.lu/IDAREJars/IDARE.jar">here</a>), load the <em>E.coli</em> core network by selecting
File -&gt; Import -&gt; Network -&gt; File and select the <em>E.Coli</em> core file you downloaded.</p>

<h4>
<a id="12-generating-an-initial-layout" class="anchor" href="#12-generating-an-initial-layout" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>1.2 Generating an initial Layout</h4>

<p>From the Layout menu, select a layout you want to apply (for this tutorial we assume, that the <strong>y-Files organic layout</strong> was choosen, or no additional layout if you have cy3sbml installed).</p>

<h4>
<a id="13-setting-up-the-network-for-idare" class="anchor" href="#13-setting-up-the-network-for-idare" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>1.3 Setting up the Network for IDARE</h4>

<p>Right click on some empty space in the network view.<br>
Select 'Apps' -&gt; 'Setup Network For IDARE'<br>
A popup will appear that lets you select the properties to be used in IDARE.<br>
On the left you can select the columns you want to use for the setup. On the right, depending on your choice of columns, you can select the values to be used for compound and interaction.<br>
Select the sbml type column as column to determine node types and the sbml id column as column to determine the node names.<br>
As compound node value select species, and as interaction node value select reaction.<br>
Check the 'Overwrite existing values' checkbox.<br>
If you are using cy3sbml please select the Base network view (it should have 187 nodes and 380 edges).</p>

<h4>
<a id="14-changing-to-the-idare-visual-style" class="anchor" href="#14-changing-to-the-idare-visual-style" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>1.4 Changing to the IDARE Visual Style</h4>

<p>From the 'Style' tab in the 'Control Panel' select the 'IDARE Visual Style' and apply it to the network. Note, that the IDARE images cannot be removed from this style. If you want to use the images with another style, you can add and remove them by right-clicking in the display area, and select Apps-&gt;Add/Remove IDARE Images</p>

<h4>
<a id="15-adding-sbml-annotations-and-gene-nodes" class="anchor" href="#15-adding-sbml-annotations-and-gene-nodes" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>1.5 Adding SBML Annotations and Gene Nodes</h4>

<p>Right click on some empty space in the network view.<br>
Select 'Apps' -&gt; 'Add SBML Annotations'<br>
Select the <em>E.Coli</em> core network file again. If you have cy3sbml this step is not necessary, as IDARE will automatically use the SBML structure provided by cy3sbml for the network.<br>
You will be asked, whether you want to add Gene Nodes. Tick the box (if not done already) and click Ok.<br>
You will notice, that additional nodes have been created, which represent the protein and gene nodes that were created. Proteins are created for all conjunctive GPR clauses (i.e. all possible and combinations that fulfil a GPR).<br>
If the SBML contains enzyme species (annotated by the "isEncodedBy" bio qualifier), you would be asked which labeling pattern (i.e. database) to use for the genes and proteins.</p>

<h2>
<a id="part2-automated-image-generation-and-loading-images-to-the-network" class="anchor" href="#part2-automated-image-generation-and-loading-images-to-the-network" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Part2: Automated image generation and loading images to the Network.</h2>

<p>In this example you will generate a few images based on artificial data and map it to the previously created and initialized network.
This example assumes that you have at loaded the E.coli core network and set up the network for IDARE (Steps 1 and 3 of the previous example).</p>

<h4>
<a id="load-data-into-idare" class="anchor" href="#load-data-into-idare" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Load Data into IDARE</h4>

<p>In the 'Control Panel', select the IDARE tab.<br>
Click on 'Add Dataset'<br>
In the dialog, click on 'Choose File'<br>
Select "Fold Change of Gene Expression ethanol vs no ethanol.xls" and click 'open'<br>
The 'DataSet Description' field will be updated with the file name<br>
As 'DataSet Type', select 'Array Dataset' (since this dataset contains only one sheet).<br>
Check the 'Use two column headers' option, as we only have one header.<br>
Click ok.  </p>

<p>Repeat the process with "Proteomics Time course of ethanol treated wild type.xls" and "Total relative C13 label.xls" which should both be loaded as 'Multiarray Dataset'.
The Proteomics time course only contains one sheet, so it can be added both as multiarray and array dataset. Adding it in both versions allows the selection of additional layouts (those specific to multiarray datasets and those specific to array datasets).<br>
Finally repeat the process with "Metabolomics fold change treatment vs Control.xls" which should be loaded as 'Array Dataset'.<br>
The labeling data only has one column for its IDs, thus you should uncheck the box for two column headers.
Example descriptions would be "External Metabolite amount information" and "Reaction Activity during the experiment" and "Internal Metabolite Amounts at specific stages".</p>

<h4>
<a id="22-create-the-visualisation" class="anchor" href="#22-create-the-visualisation" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>2.2 Create the Visualisation</h4>

<p>You can now select individual datasets you want to visualize. 
To do so, choose a selection of datasets from the table by ticking the "selected" boxes.<br>
In the lower left corner, (below the 'Create' and 'Preview' buttons) are indicators, how many nodes the datasets represent in total and how many shared nodes the selection contains.<br>
When creating visualisations for multiple datasets at once, there should be an overlap of the sets, otherwise parts of the generated image will always stay blank since no data is available.
However, you can of course create multiple different nodes by selecting different sets of data.
Images will only be created for nodes which are part of a selected dataset and old images will be kept unless a new image is created for the node.</p>

<h4>
<a id="23-adding-idare-images-to-other-styles" class="anchor" href="#23-adding-idare-images-to-other-styles" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>2.3 Adding IDARE images to other Styles</h4>

<p>If you did not select the IDARE visual style, but want to use a different style for your network, you can do so by right-clicking anywhere in a network view using the style you want to add your nodes to.
Select 'Apps' -&gt; 'Add IDARE Images'. Now the images will be shown for your choosen style.</p>

<h2>
<a id="part3-subnetwork-generation" class="anchor" href="#part3-subnetwork-generation" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Part3: Subnetwork Generation</h2>

<p>The first step of this this Example is independent on the first 2 Examples.
For the second step, it is necessary to have the SBML annotations added to the network (step 1.5), as otherwise the corresponding table column does not exist.
You can add the sbml annotation to the Cytosol network (C_c) after step 1 of this example, but this would lead to gene nodes not being added in the external compartment subnetwork.</p>

<h4>
<a id="31-create-networks-for-the-external-compartment-and-the-cytosol" class="anchor" href="#31-create-networks-for-the-external-compartment-and-the-cytosol" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>3.1 Create Networks for the External compartment and the cytosol</h4>

<p>We first want to split the network at the transporters which translocate metabolites from the external compartment to the cytosol, and vice versa.<br>
To do so:<br>
Select 'Apps' -&gt; 'IDARE' -&gt; 'Create Subnetworks'<br>
or<br>
Right Click in an empty space in the network you want to create subnetworks in -&gt; 'Apps' -&gt; 'Create Subnetworks'<br>
In the resulting dialog, you can select the column to determine the node types as well as the node names.<br>
If the network is set up for IDARE (see Part 1), appropriate columns are chosen automatically (i.e. if you have set up the network for idare - Step 1.3 - you can just accept the selection) .<br>
You further have to choose a value for nodes which form the branching points between subnetworks, and a value for nodes that form the "base" of the subnetwork.
Since the aim is to branch at the transporters between compartments reactions will be used as branching nodes and species as subnetwork nodes.<br>
Click accept, when you finished your selection.<br>
In the next dialog, you can select the column that contains the subnetwork identifiers as well as the type of layout that should be used for the subnetwork (or the "keep layout" option, if the current layout shoudl be kept)
The table allows you to select nodes, which should not branch and those which should be removed entirely  (e.g. nodes with too many connections).<br>
Depending on the network size some nodes are already suggested by the tool.<br>
Since we want to create the networks for the cytosol and external compartment, we select "sbml compartment" as column to determine sub-networks.<br>
We leave the 'Keep Layout' option as it is and select remove for the Biomass reaction (as this reaction mainly leads to a hairball structure).<br>
We also select C_c, the cytosol, and C_e, the external medium, in the sub-networks to be generated table.<br>
If we now click on the "network" tab in the 'Control Panel' we will see two new networks that were generated.   </p>

<h4>
<a id="32-create-pathway-networks-in-the-cytosol" class="anchor" href="#32-create-pathway-networks-in-the-cytosol" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>3.2 Create pathway networks in the Cytosol</h4>

<p>There is not much to see in the external compartment, as it only contains the transporters and the cytosol is still a hairball. 
To get a better overview, we would like to generate pathway networks in the cytosol.<br>
To do so, again, select 'Apps' -&gt; 'SubNetworkGenerator'<br>
However, this time, since reactions belong to pathways and metabolites can be shared between different pathways, we select species as values for branching nodes and reaction as value for subnetwork nodes.<br>
By default, the "COBRA_SUBSYSTEM" column generated by the SBMLAnnotator is selected, if available, so this selection is already fine.<br>
From the metabolites we would like to exclude currency metabolites and most cofactors.<br>
For larger networks, the pre-selection is better, but on small networks (like the e.coli core), the relative amounts of presence tend to be higher, so we have to select additional nodes to remove during the generation process.<br>
Select M_h_c, M_h2o_c, M_atp_c, M_pi_c, M_pi_c, M_nadh_c, M_nad_c, M_adp_c, M_nadph_c, M_co2_c, M_nadp_c and M_coa_c, as metabolites to remove.  </p>

<p>As pathways, we would like to create 'Glycolysis', the 'Citric Acid Cycle', 'Glyoxylate Shunt' and the 'Pentose phosphate pathway'.<br>
Click ok.  </p>

<p>You can now navigate through the different pathways. If you wish to create additional pathways, select the source network again and repeat the steps.
New links will be added to the existing networks.</p>

<p>You will notice, that links from parental networks will be included in the newly created networks, but no links are generated in these parental networks. 
In general, links will be generated for all networks built on the same column, that share linking nodes. 
If you create additional subnetworks at a later stage, links to and from old networks will be created for all newly generated networks.</p>

<p>You can see, that the labelling for ketoglutarate is much smaller than the labelling of any other measured metabolite. One potential reason is an increase in the glyoxylate shunt (visible in both the proteomics and gene expression of the enzymes involved in that pathway). </p>

<h2>
<a id="citations" class="anchor" href="#citations" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Citations</h2>

<p>[1]Horinouchi T, Tamaoka K, Furusawa C, Ono N, Suzuki S, Hirasawa T, Yomo T, Shimizu H (2010) Transcriptome analysis of parallel-evolved escherichia coli strains under ethanol stress. BMC Genomics 11(1):579, , URL <a href="http://dx.doi.org/10.1186/1471-2164-11-579">http://dx.doi.org/10.1186/1471-2164-11-579</a><br>
[2]Wang J, Chen L, Tian X, Gao L, Niu X, Shi M, Zhang W (2013) Global metabolomic and network analysis of escherichia coli responses to exogenous biofuels. Journal of Proteome Research 12(11):5302–5312, , URL <a href="http://dx.doi.org/10.1021/pr400640u">http://dx.doi.org/10.1021/pr400640u</a>, pMID: 24016299<br>
[3] Goodarzi H, Bennett BD, Amini S, Reaves ML, Hottes AK, Rabinowitz JD, Tavazoie S (2010) Regulatory and metabolic rewiring during laboratory evolution of ethanol tolerance in e. coli. Molecular Systems Biology 6(1):378–n/a, , URL <a href="http://dx.doi.org/10.1038/msb.2010.33">http://dx.doi.org/10.1038/msb.2010.33</a>, 378<br>
[4]Soufi B, Krug K, Harst A, Macek B (2015) Characterization of the e. coli proteome and its modifications during growth and ethanol stress. Frontiers in Microbiology 6:103, , URL <a href="http://journal.frontiersin.org/article/10.3389/fmicb.2015.00103">http://journal.frontiersin.org/article/10.3389/fmicb.2015.00103</a>  </p>
      </section>
      <footer>
        <p>This project is maintained by <a href="https://github.com/sysbiolux">sysbiolux</a></p>
        <p><small>Hosted on GitHub Pages &mdash; Theme by <a href="https://github.com/orderedlist">orderedlist</a></small></p>
      </footer>
    </div>
    <script src="javascripts/scale.fix.js"></script>
    
  </body>
</html>
